src/main/java/io/appium/java_client/remote/SupportsLocation.java

```diff
@@ -16,19 +16,54 @@
 
 package io.appium.java_client.remote;
 
+import io.appium.java_client.CommandExecutionHelper;
+import io.appium.java_client.ExecutesMethod;
+import io.appium.java_client.Location;
+import io.appium.java_client.MobileCommand;
 import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.html5.Location;
 import org.openqa.selenium.html5.LocationContext;
 import org.openqa.selenium.remote.html5.RemoteLocationContext;
 
-public interface SupportsLocation extends WebDriver, LocationContext {
-    public RemoteLocationContext getLocationContext();
+import java.util.Map;
 
+public interface SupportsLocation extends WebDriver, ExecutesMethod, LocationContext {
+
+    @Deprecated(forRemoval = true)
+    RemoteLocationContext getLocationContext();
+
+    /**
+     * Gets the physical location.
+     *
+     * @return A {@link Location} containing the location information. Returns null if the location is not available.
+     */
     default Location location() {
-        return getLocationContext().location();
+        Map<String, Number> result = CommandExecutionHelper.execute(this, MobileCommand.GET_LOCATION);
+        if (result == null) {
```

Comment:
let it throw an exception

Reply:
having current approach it could be a breaking change: Selenium client returns `null`

Reply:
> Selenium client returns null

This is [unfortunate](https://maximegel.medium.com/never-return-null-again-81670a10bd9b).
Lets just have our own implementation with poker and cabaret.